core/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java

```diff
@@ -188,6 +207,28 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static MultiSearchResponse fromXContext(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    static MultiSearchResponse.Item itemFromXContent(XContentParser parser) throws IOException {
+        // TODO: Is there a better way? The msearch item format is very tricky...
```

Comment:
can you elaborate on what is tricky? Is that because items can either be errors or proper instance of Item ?

Reply:
If the only key is error then we need we need to parse the xcontent is an error an otherwise parse as a search response. In order to read the the first field we need to parse it up until the field FIELD token is found to decide what to do, but then we are already to far to use `SearchResponse#fromXContent(...)`, it requires that the first token is a start object token.

Reply:
gotcha. shall we have an innerFromXContent that does not require start_object then, and accepts that we are already at the field_name token? That could be called from msearch? Exceptions should be parseable already with this technique.